src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -1081,25 +1094,96 @@ impl_install_tree (RpmOstreeTreeComposeContext *self,
+          if (!rojig_name)
+            return glnx_throw (error, "No `rojig` entry in manifest");
+
+          g_autoptr(GKeyFile) tsk = g_key_file_new ();
+          const char *rojig_output_repo_id = "rpmostree-rojig-output";
+
+          g_autofree char *rojig_spec = g_strconcat (rojig_output_repo_id, ":", rojig_name, NULL);
+          g_key_file_set_string (tsk, "tree", "rojig", rojig_spec);
+          { const char *repos[] = { rojig_output_repo_id, };
+            g_key_file_set_string_list (tsk, "tree", "repos", (const char*const*)repos, 1);
           }
-        else
-          {
-            g_propagate_error (error, g_steal_pointer (&temp_error));
+
+          g_autoptr(RpmOstreeTreespec) treespec = rpmostree_treespec_new_from_keyfile (tsk, error);
+          if (!treespec)
+            return FALSE;
+          g_autoptr(RpmOstreeContext) corectx = rpmostree_context_new_tree (self->cachedir_dfd, self->repo, cancellable, error);
+          if (!corectx)
             return FALSE;
+          DnfContext *dnfctx = rpmostree_context_get_dnf (corectx);
+          if (!glnx_shutil_mkdir_p_at (self->workdir_dfd, "rojig-repos", 0755, cancellable, error))
+            return FALSE;
+          { g_autofree char *repopath = g_strconcat ("rojig-repos/", rojig_output_repo_id, ".repo", NULL);
+            g_autofree char *repo_contents = g_strdup_printf ("[%s]\n"
+                                                              "baseurl=file://%s\n"
+                                                              "gpgcheck=0\n",
+                                                              rojig_output_repo_id,
+                                                              rojig_outputdir);
+            if (!glnx_file_replace_contents_at (self->workdir_dfd, repopath,
+                                                (guint8*)repo_contents, -1, 0,
+                                                cancellable, error))
+              return FALSE;
           }
-      }
-    else
-      g_print ("Previous commit: %s\n", self->previous_checksum);
-  }
+          g_autofree char *reposdir_abspath = glnx_fdrel_abspath (self->workdir_dfd, "rojig-repos");
+          dnf_context_set_repo_dir (dnfctx, reposdir_abspath);
+          if (!rpmostree_context_setup (corectx, NULL, NULL, treespec, cancellable, error))
+            return FALSE;
+          if (!rpmostree_context_prepare_rojig (corectx, cancellable, error))
+            return FALSE;
+          DnfPackage *rojig_pkg = rpmostree_context_get_rojig_pkg (corectx);
+          g_print ("Previous rojig: %s\n", dnf_package_get_nevra (rojig_pkg));
+          self->previous_checksum = g_strdup (rpmostree_context_get_rojig_inputhash (corectx));
+          self->previous_version = g_strdup (dnf_package_get_version (rojig_pkg));
+          self->previous_inputhash = g_strdup (rpmostree_context_get_rojig_inputhash (corectx));
+        }
+      else
```

Comment:
Why not just drop `--ex-rojig-init` and enhance the core so we can discern the `ENOENT` case from other errors? Then it'd be fully symmetrical with the `--repo` case.